pkg/cache/partition_info.go

```diff
@@ -477,12 +477,17 @@ func (pi *PartitionInfo) releaseAllocationsForApplication(toRelease *commonevent
 	for _, alloc := range allocationsToRelease {
 		// remove allocation from node
 		node := pi.nodes[alloc.AllocationProto.NodeID]
+		// node could be nil if when release happens, the node is already
+		// removed from the cluster. in this case, we need to make sure
+		// resources are released correctly. otherwise these resources will
+		// be leaked.
 		if node == nil || node.GetAllocation(alloc.AllocationProto.UUID) == nil {
-			log.Logger().Info("node is not found for allocation",
-				zap.Any("allocation", alloc))
-			continue
+			log.Logger().Info("node is not found or the allocation is not found",
+				zap.String("node", alloc.AllocationProto.NodeID),
+				zap.String("allocationID", alloc.AllocationProto.UUID))
+		} else {
+			node.RemoveAllocation(alloc.AllocationProto.UUID)
 		}
-		node.RemoveAllocation(alloc.AllocationProto.UUID)
 		totalReleasedResource.AddTo(alloc.AllocatedResource)
```

Comment:
This needs to be reverted too. If we have a nil node we need to continue. It is almost an error case but we can only update if we have a node and an alloc.
The queue will get out of sync if we remove the alloc while the node is nil. The nil node is a result of the queue already being updated.

Reply:
Sorry, I missed this one earlier. Just done in the latest commit.